gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -143,8 +144,9 @@
 
   protected final FileSystem fs;
   private final List<Service> services = Lists.newArrayList();
-  private final String applicationName;
-  private final String applicationId;
+  protected final String applicationName;
+  protected final String applicationId;
+  protected final Optional<ContainerId> containerId;
```

Comment:
Are we spilling implementation details (e.g. yarn for here) into a generic-purpose class (`GobblinTaskRunner`) by doing this?

Reply:
Agreed.